spyder/plugins/outlineexplorer/api.py

```diff
@@ -262,3 +228,72 @@ def get_block_number(self):
             # Avoid calling blockNumber if not a valid block
             return None
         return self.block.blockNumber()
+
+
+def document_cells(block, forward=True):
```

Comment:
Please move these functions to the beginning of the file because they are auxiliary functions, used here and by other classes around our codebase.